modules/api/src/main/java/org/apache/ignite/tx/StaleSchemaRollbackException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.tx;
+
+import java.util.UUID;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Thrown when, during an attempt to execute a commit, it turns out that the schema at the moment of commit is not compatible
+ * with the schema at the moment when the transaction was started and, hence, the transaction was aborted instead.
+ */
+public class StaleSchemaRollbackException extends RollbackException {
```

Comment:
This looks very similar to `IncompatibleSchemaException`, is there any significant difference? Can we keep only one of those exceptions?

Reply:
`StaleSchemaRollbackException` means that an attempt to commit failed and the transaction was actually aborted. This seems to be an important piece of information. The user could do `catch (RollbackException)` to catch any 'could not commit, instead aborted', including the case when it happened due to an incompatible schema change.

On the other hand, `IncompatibleSchemaException` is not about a commit attempt.